test/controllers/v1/ReportingUnitRoutingSpec.scala

```diff
@@ -0,0 +1,141 @@
+package controllers.v1
+
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{ FreeSpec, Matchers }
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.Application
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import support.matchers.HttpServerErrorStatusCodeMatcher.aServerError
+
+/*
+ * We are relying on the router to perform argument validation for us (via regex constraints).
+ * This spec tests that the router is configured correctly.
+ *
+ * Because we are relying on the router to validate arguments, and the simple routes file configuration does not
+ * compose, each and every route must be tested individually.  In addition, some of the regex are non-trivial,
+ * and as the saying goes:
+ *
+ *     "Some people, when confronted with a problem, think 'I know, I'll use regular expressions.'
+ *     Now they have two problems."
+ *
+ * For example, in order to fully test the period regex, we test that each and every possible month is considered
+ * valid.  These are downsides of this "router based validation" approach ...
+ */
+class ReportingUnitRoutingSpec extends FreeSpec with Matchers with GuiceOneAppPerSuite with ScalaFutures {
```

Comment:
Would it be possible to make the router specs more generic, as only the validation of the id changes?

Reply:
Possibly.  I'm not planning to address this now, but we can look to revisit this as we progress - I'm open to suggestions as to how we can better structure this.